pkgs/development/python-modules/textual-autocomplete/default.nix

```
@@ -0,0 +1,38 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, poetry-core
+, textual
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "textual-autocomplete";
+  version = "3.0.0-unstable-2024-07-10";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "darrenburns";
+    repo = "textual-autocomplete";
+    rev = "bbacfa9";
+    hash = "sha256-m2ATH2BNoVCoEzKb5xxe4KPvUlfrwfE+widRIdApkL8=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    textual
+    typing-extensions
+  ];
+
+  pythonImportsCheck = [ "textual_autocomplete" ];
+
```

Could you enable tests with `pytestCheckHook`?

> Make sure the tests are enabled using for example [pytestCheckHook](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook) and, in the case of libraries, are passing for all interpreters. If certain tests fail they can be disabled individually. Try to avoid disabling the tests altogether. In any case, when you disable tests, leave a comment explaining why.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines

---

Neither `textual-autocomplete` nor `posting` have tests implemented. Pytest returns an error when no test is present and this breaks the build process.
See: https://github.com/darrenburns/textual-autocomplete/tree/version-two/tests

---

You may add this instead then
```suggestion
  doCheck = false; # no tests
```